src/index.ts

```diff
@@ -178,3 +189,24 @@ function getPreloadLink(node: any) {
 
   return link;
 }
+
+function appendInlineTextIfNeeded(
+  pendingText: Text | null,
+  inlineTextHostNode: Node | null
+) {
+  if (pendingText && inlineTextHostNode) {
+    inlineTextHostNode.appendChild(pendingText);
+  }
+}
+
+function isInlineHost(node: Node | null) {
+  if (node?.nodeType !== Node.ELEMENT_NODE) {
```

Comment:
Could be mistaken, but isn't `node` guaranteed to be defined here? We could skip the null check if so

Reply:
ah yeah, sorry my bad 🙇 , I've added a non-null assertion instead in the caller